Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java

```diff
@@ -62,7 +56,27 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
     @Override
     public void onResume() {
         super.onResume();
-        payPalLauncher.handleReturnToAppFromBrowser(requireContext(), requireActivity().getIntent());
+        PayPalPendingRequest.Started pendingRequest = getPendingRequest();
+        if (pendingRequest != null) {
+            PayPalPaymentAuthResult paymentAuthResult = payPalLauncher.handleReturnToAppFromBrowser(pendingRequest, requireActivity().getIntent());
+            clearPendingRequest();
+            if (paymentAuthResult != null) {
+                completePayPalFlow(paymentAuthResult);
+            } else {
+                handleError(new Exception("User did not complete payment flow"));
+            }
+        }
+    }
+
+    private void storePendingRequest(PayPalPendingRequest.Started request) {
+        PendingRequestStore.putPayPalPendingRequest(requireContext(), request);
```

Comment:
```suggestion
        PendingRequestStore.getInstance(requireContext()).putPayPalPendingRequest(request);
```
nit: Singleton accessor.

Reply:
Good call - fixed